File: cortex/domain/routing.py
from __future__ import annotations

import os

HUMAN_RECIPIENT = "suren"
LEGACY_HUMAN_RECIPIENT = "human"

RESERVED_NAMES: frozenset[str] = frozenset({HUMAN_RECIPIENT, LEGACY_HUMAN_RECIPIENT})

HUMAN_SENDER_TYPE = HUMAN_RECIPIENT
AGENT_SENDER_TYPE = "agent"
SYSTEM_SENDER_TYPE = "system"

DEPRECATED_HUMAN_WARNING = (
    f"to={LEGACY_HUMAN_RECIPIENT!r} is deprecated — use to={HUMAN_RECIPIENT!r}"
)

UNKNOWN_ACTOR = "unknown"


def canonical_recipient(recipient: str) -> tuple[str, str | None]:
    if recipient == LEGACY_HUMAN_RECIPIENT:
        return HUMAN_RECIPIENT, DEPRECATED_HUMAN_WARNING
    return recipient, None


def is_human_recipient(recipient: str) -> bool:
    return recipient in (HUMAN_RECIPIENT, LEGACY_HUMAN_RECIPIENT)


def resolve_actor() -> str:
    session_name = os.environ.get("CORTEX_SESSION_NAME")
    if session_name:
        return session_name
    actor = os.environ.get("CORTEX_ACTOR")
    if actor:
        return actor
    return UNKNOWN_ACTOR


def sender_type_for(actor: str) -> str:
    return HUMAN_SENDER_TYPE if actor == HUMAN_RECIPIENT else AGENT_SENDER_TYPE

File: src/channels-mcp/routing.ts
export const HUMAN_RECIPIENT = "suren";
export const LEGACY_HUMAN_RECIPIENT = "human";

export const RESERVED_NAMES: ReadonlySet<string> = new Set([
  HUMAN_RECIPIENT,
  LEGACY_HUMAN_RECIPIENT,
]);

export const HUMAN_SENDER_TYPE = HUMAN_RECIPIENT;
export const AGENT_SENDER_TYPE = "agent";
export const SYSTEM_SENDER_TYPE = "system";

export const DEPRECATED_HUMAN_WARNING =
  `to="${LEGACY_HUMAN_RECIPIENT}" is deprecated — use to="${HUMAN_RECIPIENT}"`;

export function canonicalRecipient(
  recipient: string
): { canonical: string; warning: string | null } {
  if (recipient === LEGACY_HUMAN_RECIPIENT) {
    return { canonical: HUMAN_RECIPIENT, warning: DEPRECATED_HUMAN_WARNING };
  }
  return { canonical: recipient, warning: null };
}

export function isHumanRecipient(recipient: string): boolean {
  return recipient === HUMAN_RECIPIENT || recipient === LEGACY_HUMAN_RECIPIENT;
}
